### PR TITLE
Remove custom updater

### PR DESCRIPTION
**Pull request by aaemnnosttv**
## Summary

**Do not merge until preparing for 1.0.0 release**

Addresses issue #644 

## Relevant technical choices

* Required changing the plugin's slug in E2E test usage with `(activate|deactivate)Plugin()` from `google-site-kit` to `site-kit-by-google` for consistency with how WordPress generates a plugin's slug for use on `plugins.php` which is title-based rather than directory name-based. This was overridden by the `Updater` filtering the plugin's information which caused the `plugin-activation` test to fail once removed.

## Checklist

- [x] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 4.7 and PHP 5.4.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

### CODE REVIEW

**Comment by aaemnnosttv** [comment]
@felixarntz Leaving this as a draft for now to prevent merging, but changes are ready to go.

File: tests/e2e/specs/plugin-activation.test.js [hunk 1]
```diff
@@ -5,27 +5,27 @@ import { deactivatePlugin, activatePlugin } from '@wordpress/e2e-test-utils';
 
 describe( 'Plugin Activation Notice', () => {
 	beforeEach( async () => {
-		await deactivatePlugin( 'google-site-kit' );
+		await deactivatePlugin( 'site-kit-by-google' );
```
Comment (felixarntz) [inline tests/e2e/specs/plugin-activation.test.js]: @aaemnnosttv Why did you change the slug here in this file?

Comment (felixarntz) [reply tests/e2e/specs/plugin-activation.test.js]: Summarizing our Slack conversation, because WordPress only uses the proper slug if it finds that plugin in an update request, let's check whether a plugin with slug `google-site-kit` is available in the list, and otherwise fall back to `site-kit-by-google`.

As soon as the plugin is launched, the latter should no longer occur, so we can remove that at some point.
